internal/proxy/task_query.go

```
@@ -493,9 +493,13 @@ func (t *queryTask) PreExecute(ctx context.Context) error {
 	t.RetrieveRequest.IsIterator = queryParams.isIterator
 
 	if collectionInfo.collectionTTL != 0 {
-		physicalTime, _ := tsoutil.ParseTS(guaranteeTs)
+		physicalTime := tsoutil.PhysicalTime(t.GetBase().GetTimestamp())
 		expireTime := physicalTime.Add(-time.Duration(collectionInfo.collectionTTL))
 		t.CollectionTtlTimestamps = tsoutil.ComposeTSByTime(expireTime, 0)
+		// preventing overflow, abort ttl timestamp
+		if t.CollectionTtlTimestamps > t.GetBase().GetTimestamp() {
+			t.CollectionTtlTimestamps = 0
```

We are not likely to go here after the change, but can we throw an error instead of "correct" the error silently?

---

throwing an error is doable. however user could do nothing about this if the corner case happened?